### DISCUSSION TITLE
Nginx reverse proxy

### DISCUSSION THREAD

**Discussion by builderjer** [roflcoopter/viseron#849] [Q&A] [answered]
I am trying to setup a reverse proxy with nginx and certbot.  I saw the closed issue [here](https://github.com/roflcoopter/viseron/issues/666), and my nginx config is very similar.

```
server {
	server_name <myhostname>;

	location / {
                proxy_pass "http://<my_local_ip_address>:<my_local_port>";
        }


    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/********-0001/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/********-0001/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
server {
    if ($host = <myhostname>) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


	server_name <myhostname>;
    listen 80;
    return 404; # managed by Certbot
}
```

From my local LAN, everything works as expected, but when I try from my proxy, I can login correctly, but no cameras are shown, and I cannot get into the settings and such.

I love this project, and hope you can help me sort this out.

EDIT:

This is showing in my logs
```
viseron  | [2024-12-08 08:49:03] [WARNING ] [tornado.access] - 400 GET /websocket (<my_local_ip>) 1.42ms
```

**Comment by roflcoopter** [comment, marked as answer]
I believe this could be the explanation: https://www.f5.com/company/blog/nginx/websocket-nginx

Does this config work?
```
server {
	server_name <myhostname>;

	location / {
                proxy_pass "http://<my_local_ip_address>:<my_local_port>";
        }


    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/********-0001/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/********-0001/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
    proxy_set_header Host $host;

}
server {
    if ($host = <myhostname>) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


	server_name <myhostname>;
    listen 80;
    return 404; # managed by Certbot
}
```

**Reply by builderjer** [reply]
Awesome!!!  And thank you for your quick response.

Yes this config works great!!

**Reply by neagix** [reply]
Would it be possible to make it work with a non-root location?
Something like a base URI or the `X-Forwarded-For` header support.

Right now it does not work when location is not `/` (not even using a `rewrite` hack).

**Reply by neagix** [reply]
Opened a PR [here](https://github.com/roflcoopter/viseron/pull/1136) to add support.

**Reply by roflcoopter** [reply]
> Opened a PR [here](https://github.com/roflcoopter/viseron/pull/1136) to add support.

Thank you, will take a look soon during the week!
